teos/cli/teos_cli.py

```diff
@@ -14,37 +15,65 @@
 
 from teos import DEFAULT_CONF, DATA_DIR, CONF_FILE_NAME
-from teos.cli.help import show_usage, help_get_all_appointments
+from teos.cli.help import (
+    show_usage,
+    help_get_all_appointments,
+    help_get_tower_info,
+    help_get_users,
+    help_get_user,
+)
 from teos.protobuf.tower_services_pb2_grpc import TowerServicesStub
+from teos.protobuf.user_pb2 import GetUserRequest
 
 
-def get_all_appointments(rpc_host, rpc_port):
+def to_json(obj):
+    return json.dumps(obj, indent=4)
+
+
+def formatted(func):
+    """Transforms the given function by wrapping the return value with json_format.MessageToDict followed by
+    json.dumps, in order to print the result in a prettyfied json format.
     """
-    Gets information about all appointments stored in the tower.
 
-    Args:
-        rpc_host (:obj:`str`): the hostname (or IP) where the rpc server is running.
-        rpc_port (:obj:`int`): the port where the rpc server is running.
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+        result_dict = json_format.MessageToDict(
+            result, including_default_value_fields=True, preserving_proto_field_name=True
+        )
+        return to_json(result_dict)
+
+    return wrapper
+
 
-    Returns:
-        :obj:`dict` a dictionary containing all the appointments stored by the Responder and Watcher if the tower
-        responds.
+class RPCClient:
+    """Creates and keeps a connection to the an RPC serving TowerServices. It has methods to call each of the
+    available grpc services, and it returns a pretty-printed json response.
+    Errors from the grpc calls are not handled.
     """
 
-    try:
-        with grpc.insecure_channel(f"{rpc_host}:{rpc_port}") as channel:
-            stub = TowerServicesStub(channel)
-            r = stub.get_all_appointments(Empty())
-            response = json_format.MessageToDict(
-                r.appointments, including_default_value_fields=True, preserving_proto_field_name=True
-            )
+    def __init__(self, rpc_host, rpc_port):
+        self.rpc_host = rpc_host
+        self.rpc_port = rpc_port
+        self.channel = grpc.insecure_channel(f"{rpc_host}:{rpc_port}")
+        self.stub = TowerServicesStub(self.channel)
+
+    @formatted
+    def get_all_appointments(self):
+        result = self.stub.get_all_appointments(Empty())
+        return result.appointments
 
-        return response
+    @formatted
+    def get_tower_info(self):
+        return self.stub.get_tower_info(Empty())
 
-    # FIXME: Handle different errors
-    except grpc.RpcError:
-        print("Can't connect to the Eye of Satoshi. RPC server cannot be reached", file=sys.stderr)
-        return None
+    def get_users(self):
+        result = self.stub.get_users(Empty())
+        return to_json(list(result.user_ids))
+
+    @formatted
+    def get_user(self, user_id):
+        return self.stub.get_user(GetUserRequest(user_id=user_id), indent=4)
```

Comment:
I don't think that `indent=4` belongs there. Also, the return of this method looks as follows:

```
{
    "user": {
        "subscription_expiry": 4433.0,
        "appointments": {
            "421bc1a536a7e6c4d9138110b9421e54881c86ad": 1.0
        },
        "available_slots": 99.0
    }
}
```

I'd either get rid of the external dictionary (user: {}) or change it so intead of user it shows the id. e.g:

```
{
    0207213bde4b80da7e58bfff65cd0d3c4d7d349461595a534ea9949d2059d382ee: {
        "subscription_expiry": 4433.0,
        "appointments": {
            "421bc1a536a7e6c4d9138110b9421e54881c86ad": 1.0
        },
        "available_slots": 99.0
    }
}
```

I'd lean towards the first tho.

Reply:
Also, the return contains the number of slots taken by every appointment:

"421bc1a536a7e6c4d9138110b9421e54881c86ad": **1.0**

I don't think that's relevant information.

Reply:
I was just dumping the whole UserInfo object; instead, now I'm explicitly taking the `subscription_expiry` and `available_slots` fields, and only putting the keys of the "appointments" dictionary as an array, so it looks like this after flattening:

```
{
    "subscription_expiry": 4320.0,
    "appointments": [],
    "available_slots": 100.0
}
```

(don't the floats, will be taken care of)

Done in 8171b65ea7350877a911b0f95f9a95cae00f1441